### PR TITLE
Update ToggleButtons constraints default and add new constraints parameter

### LINKED ISSUE #39216: Add ToggleButtons.constraints parameter
Currently, there is no way to control the constraints of each button in the ToggleButtons widget. It currently defaults to `const BoxConstraints(minWidth: 88.0, minHeight: 36.0)`, which is the constraints for `RawMaterialButton`, but this is undesirable if we want it to look like the square toggle buttons:
image

### PR DESCRIPTION
**Pull request by shihaohong**
## Description

Currently, there is no way to control the constraints of each button in the ToggleButtons widget. It currently defaults to const `BoxConstraints(minWidth: 88.0, minHeight: 36.0)`, which is the constraints for RawMaterialButton, but this is undesirable if we want it to look like the square toggle buttons. 

image

This PR does the following: 
1) Sets ToggleButtons' default BoxConstraints to BoxConstraints(minWidth: 48.0, minHeight: 48.0)
2) Adds a `constraints` parameter to ToggleButtons and ToggleButtonsTheme

## Required Follow-up
- [x] Update flutter/assets-for-api-docs to ensure that the defaults in the code sample are updated

## Related Issues

Fixes https://github.com/flutter/flutter/issues/39216

## Tests

I added the following tests:

1) Test for ToggleButtons default and custom constraints
2) Test for custom ToggleButtonsTheme
3) Updated existing test to account for new `constraints` parameter

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [x] Yes, this is a breaking change (Please read [Handling breaking changes]). - TODO: send email and link here once LGTM is received
- [ ] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/material/toggle_buttons.dart [hunk 1]
```diff
@@ -784,13 +800,15 @@ class _ToggleButton extends StatelessWidget {
     }
 
     final TextStyle currentTextStyle = textStyle ?? toggleButtonsTheme.textStyle ?? theme.textTheme.body1;
+    final BoxConstraints currentConstraints = constraints ?? toggleButtonsTheme.constraints ?? const BoxConstraints(minWidth: 48.0, minHeight: 48.0);
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/toggle_buttons.dart]: You can use [kMinInteractiveDimension](https://api.flutter.dev/flutter/material/kMinInteractiveDimension-constant.html) here

File: packages/flutter/lib/src/material/toggle_buttons.dart [hunk 2]
```diff
@@ -223,6 +224,14 @@ class ToggleButtons extends StatelessWidget {
   /// are active, selected, or disabled.
   final TextStyle textStyle;
 
+  /// Defines the button's size.
+  ///
+  /// Typically used to constrain the button's minimum size.
+  ///
+  /// If this property is null, then
+  /// BoxConstraints(minWidth: 48.0, minHeight: 48.0) is be used.
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/toggle_buttons.dart]: Although the 48 is kMinInteractiveDimension I think the documentation is easier to understand the way you have it.

**Review by HansMuller — APPROVED** [review]
LGTM

**Comment by ShindongLee** [comment]
Hi. Thanks for your PR but when can i use it? I'm using Flutter 1.10.3-pre.66 which is latest in master channel but still not seeing a 'constraints' parameter in ToggleButtons T_T

**Pull request merged**